File: win/directx/sdk-9.0/Samples/VB.Net/DirectPlay/Tutorials/Tut09_ClientServer/Client/HostInfo.vb
'----------------------------------------------------------------------------
' File: HostInfo.cs
'
' Desc: Contains the HostInfo collection item class. Objects of this class
'       are used to store information about detected hosts.
'
' Copyright (c) Microsoft Corp. All rights reserved.
'----------------------------------------------------------------------------- 
Imports System
Imports Microsoft.DirectX.DirectPlay


Namespace Tut09_Client
    '/ <summary>
    '/ This class contains display and address information for a detected host.
    '/ </summary>
    Public Class HostInfo
        Public GuidInstance As Guid ' Instance Guid for the session
        Public HostAddress As Address ' DirectPlay address of the session host  
        Public SessionName As String
        ' Display name for the session
        '/ <summary>
        '/ Used by the system collection class
        '/ </summary>
        Public Overloads Function Equals(ByVal obj As Object) As Boolean
            Dim node As HostInfo = CType(obj, HostInfo)
            Return GuidInstance.Equals(node.GuidInstance)
        End Function 'Equals

        '/ <summary>
        '/ Used by the system collection class
        '/ </summary>
        Public Overrides Function GetHashCode() As Integer
            Return GuidInstance.GetHashCode()
        End Function 'GetHashCode


        '/ <summary>
        '/ Used by the system collection class
        '/ </summary>
        Public Overrides Function ToString() As String
            Dim displayString As String = IIf(Not (SessionName Is Nothing), SessionName, "<unnamed>")
            displayString += " (" + HostAddress.GetComponentString("hostname")
            displayString += ":" + HostAddress.GetComponentInteger("port").ToString() + ")"

            Return displayString
        End Function 'ToString
    End Class 'HostInfo
End Namespace 'Tut09_Client

File: win/directx/sdk-9.0/Samples/C#/DirectPlay/Tutorials/Tut10_ThreadPool/HostInfo.cs
//----------------------------------------------------------------------------
// File: HostInfo.cs
//
// Desc: Contains the HostInfo collection item class. Objects of this class
//       are used to store information about detected hosts.
//
// Copyright (c) Microsoft Corp. All rights reserved.
//----------------------------------------------------------------------------- 
using System;
using Microsoft.DirectX.DirectPlay;

namespace Tut10_ThreadPool
{
    /// <summary>
    /// This class contains display and address information for a detected host.
    /// </summary>
    public class HostInfo
    {
        public Guid GuidInstance; // Instance Guid for the session
        public Address HostAddress; // DirectPlay address of the session host  
        public string SessionName; // Display name for the session

        /// <summary>
        /// Used by the system collection class
        /// </summary>
        public override bool Equals( object obj ) 
        { 
            HostInfo node = (HostInfo) obj;
            return GuidInstance.Equals( node.GuidInstance );
        }

        /// <summary>
        /// Used by the system collection class
        /// </summary>
        public override int GetHashCode()
        {
            return GuidInstance.GetHashCode();
        }

        /// <summary>
        /// Used by the system collection class
        /// </summary>
        public override string ToString()
        {
            string displayString = (SessionName != null) ? SessionName : "<unnamed>";
            displayString += " (" + HostAddress.GetComponentString("hostname");
            displayString += ":" + HostAddress.GetComponentInteger("port").ToString() + ")";

            return displayString;
        }
    }
}